internal/fetchclient/fetchclient.go

```diff
@@ -41,21 +41,21 @@ type Client struct {
 }
 
 // New returns a new client.
-func New(ctx context.Context) *Client {
-	var client *http.Client
-	if ghToken := os.Getenv("GITHUB_TOKEN"); ghToken != "" {
-		ts := oauth2.StaticTokenSource(
-			&oauth2.Token{AccessToken: ghToken},
-		)
-		client = oauth2.NewClient(ctx, ts)
+func New(ctx context.Context) (context.Context, *Client) {
+	httpClient := http.DefaultClient
```

Comment:
Something I found useful in the past is checking the status of the rate limits after instantiating the client. I wonder if we're just exhausting the amount of work that a bot user can do across our repositories.

https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#checking-the-status-of-your-rate-limit

The only downside is there's no way (afaik) to check the secondary rate limits, and if we're hitting those then 🤷 .